staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -131,6 +131,11 @@ type Config struct {
 	// Version forces a specific version to be used (if registered)
 	// Do we need this?
 	// Version string
+
+	// RESTClientBuilder allows to inject function for building REST client.
+	// E.g. for mocking REST client to be used by various clientset clients.
+	// If not set, Config.RESTClient() falls to the default RESTClientFor.
+	RESTClientBuilder func(config *Config) (Interface, error)
```

Comment:
It's unclear to me how this solves the issue in the PR description. Without demonstrated use in the repo, I question whether we should add this injection capability.